habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py

```diff
@@ -203,28 +206,28 @@ def calculate_walk_pose(
             step_size = 0
 
         # Advance mocap frame
-        prev_mocap_frame = self.walk_mocap_frame
+        # prev_mocap_frame = self.walk_mocap_frame
         self.walk_mocap_frame = (
             self.walk_mocap_frame + step_size
         ) % self.walk_motion.num_poses
 
         # Compute how much distance we covered in this motion
-        prev_cum_distance_covered = self.walk_motion.displacement[
-            prev_mocap_frame
-        ]
-        new_cum_distance_covered = self.walk_motion.displacement[
-            self.walk_mocap_frame
-        ]
-
-        offset = 0
-        if self.walk_mocap_frame < prev_mocap_frame:
-            # We looped over the motion
-            offset = self.walk_motion.displacement[-1]
-
-        distance_covered = max(
-            0, new_cum_distance_covered + offset - prev_cum_distance_covered
-        )
-        dist_diff = min(distance_to_walk, distance_covered)
+        # prev_cum_distance_covered = self.walk_motion.displacement[
+        #     prev_mocap_frame
+        # ]
+        # new_cum_distance_covered = self.walk_motion.displacement[
+        #     self.walk_mocap_frame
+        # ]
+
+        # offset = 0
+        # if self.walk_mocap_frame < prev_mocap_frame:
+        #     # We looped over the motion
+        #     offset = self.walk_motion.displacement[-1]
+
+        # distance_covered = max(
+        #     0, new_cum_distance_covered + offset - prev_cum_distance_covered
+        # )
+        # dist_diff = min(distance_to_walk, distance_covered)
```

Comment:
We need those changes to set humanoid walk speed. However, this may break other stuff.

Perhaps we could make animation-driven humanoid speed optional via config? If we do that, I'm afraid that it will be hard for other developers to figure out why the speed is limited.

Ideally, animation speed should be driven by locomotion speed.

@xavierpuigf Thoughts?

Reply:
For now, enabling/disabling this through config and adding comments both at config and code-level sounds like the least effort option to still maintain "other" uses of code.

Wdyt? Also, I do not completely understand this code layer so your help to add comments here would be perfect.

Reply:
Yep! The fast solution is to disable animation. I have a way to remove the speed limitation but havent had time to get into it yet. If we want to have animation, I can work on integrating it next week.

Reply:
We should not merge this as is though, it will break the functionalities for any other kind of motion